lib/posix/posix.nim

```diff
@@ -663,6 +663,25 @@ proc waitid*(a1: cint, a2: Id, a3: var SigInfo, a4: cint): cint {.
 proc waitpid*(a1: Pid, a2: var cint, a3: cint): Pid {.
   importc, header: "<sys/wait.h>".}
 
+type Rusage* {.importc: "struct rusage", header: "<sys/resource.h>",
+               pure, final.} = object
+  ru_utime*, ru_stime*: Timeval                       # User and system time
+  ru_maxrss*, ru_ixrss*, ru_idrss*, ru_isrss*,        # memory sizes
+    ru_minflt*, ru_majflt*, ru_nswap*,                # paging activity
+    ru_inblock*, ru_oublock*, ru_msgsnd*, ru_msgrcv*, # IO activity
+    ru_nsignals*, ru_nvcsw*, ru_nivcsw*: clong        # switching activity
+
+proc wait4*(pid: Pid, status: ptr cint, options: cint, rusage: ptr Rusage): Pid
+  {.importc, header: "<sys/wait.h>", discardable.}
+
+const
+  RUSAGE_SELF* = cint(0)
+  RUSAGE_CHILDREN* = cint(-1)
+  RUSAGE_THREAD* = cint(1)    # This one is less std; Linux, BSD agree though.
+
+proc getrusage*(who: cint, rusage: ptr Rusage): cint
+  {.importc, header: "<sys/resource.h>", discardable.}
```

Comment:
"On success, zero is returned.  On error, -1 is returned, and errno is set appropriately." So this is not ``discardable``.

Reply:
Yeah, but, according to docs/kernel code, failure is only possible from user error - using a bad address or a bad `RUSUAGE_WHO` value.  That seems like a prime example of when things can be `discardable` - the user knows arguments are valid.  If the user doesn't know the params are valid they can catch & check the value, if they do they don't need to `discard`.

It's also not easy to create real bugs by not checking.  Also, you often/usually do a pair of `getrusage` calls across some activity to measure something (sort of the way one uses a time query). If both calls fail you just get a zero difference. `discardable` doesn't remove the result - it just doesn't mandate `discard` or catching it, correct?

I'm not adamant about this, but it seemed like a perfect case for `discardable` to me.